src/lxc/cgroups/cgfsng.c

```diff
@@ -411,7 +412,6 @@ static bool cg_legacy_filter_and_set_cpus(char *path, bool am_initialized)
 	if (!file_exists(__ISOL_CPUS)) {
 		/* This system doesn't expose isolated cpus. */
 		DEBUG("The path \""__ISOL_CPUS"\" to read isolated cpus from does not exist");
-		cpulist = posscpus;
```

Comment:
There's a `goto copy_parent;` a few lines below which copies from `cpulist`, so this is a behavior change as well, no?
So unless this is intentional, to deal with the `cpulist != posscpus` comparison the cleanup had before it would probably be the most clear if we add a non-autofreed `char *cpulist_to_copy;` set here, below, and finally before the `copy_parent` label?

Reply:
Good catch!

Reply:
I think we can do:
```c
	if (!isdigit(isolcpus[0])) {
		TRACE("No isolated cpus detected");
		/* No isolated cpus but we weren't already initialized by
		 * someone. We should simply copy the parents cpuset.cpus
		 * values.
		 */
		if (!am_initialized) {
			DEBUG("Copying cpu settings of parent cgroup");
			cpulist = posscpus;
			goto copy_parent;
		}
		/* No isolated cpus but we were already initialized by someone.
		 * Nothing more to do for us.
		 */
		return true;
	}
.
.
.	ret = lxc_write_to_file(fpath, cpulist, strlen(cpulist), false, 0666);
	if (cpulist == posscpus)
		cpulist = NULL;
	if (ret < 0) {
		SYSERROR("Failed to write cpu list to \"%s\"", fpath);
		return false;
	}
```